### PR TITLE
Issue/2148 Tabs / Tabs Bar

### PR DESCRIPTION
**Pull request by mhaegeleMHP**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://github.com/porscheofficial/oss-docs/blob/67c5b24838a293ce7a964884e6005eb71f2b8579/CONTRIBUTOR_LICENSE_AGREEMENT.md)

#### References

- Preview: https://designsystem.porsche.com/issue/2148/

#### Scope

Short description of the pull request scope

### CODE REVIEW

File: packages/components-js/tests/vrt/puppeteer/specs/tabs-bar.vrt.ts [hunk 1]
```diff
@@ -35,10 +35,7 @@ xit('should have no visual regression for :hover + :focus-visible', async () =>
           <a href="#">Anchor Tab Three</a>
         </p-tabs-bar>`;
 
-      await setContentWithDesignSystem(
-        page,
-        getThemedBodyMarkup(getElementsMarkup, { themes: ['light', 'dark', 'light-electric'] })
-      );
+      await setContentWithDesignSystem(page, getThemedBodyMarkup(getElementsMarkup, { themes: ['light', 'dark'] }));
```
Comment (denyo) [inline packages/components-js/tests/vrt/puppeteer/specs/tabs-bar.vrt.ts:38]: `themes` option does not exist anymore.

File: packages/components-js/tests/vrt/puppeteer/specs/tabs.vrt.ts [hunk 2]
```diff
@@ -43,7 +43,7 @@ xit('should have no visual regression for :focus-visible', async () => {
 
       await setContentWithDesignSystem(
         page,
-        getThemedBodyMarkup(getElementsMarkup, { themes: ['light', 'dark', 'light-electric'], states: ['focus'] })
+        getThemedBodyMarkup(getElementsMarkup, { themes: ['light', 'dark'], states: ['focus'] })
```
Comment (denyo) [inline packages/components-js/tests/vrt/puppeteer/specs/tabs.vrt.ts:46]: Same here.

File: packages/components/src/styles/focus-visible-fallback.ts [hunk 3]
```diff
@@ -22,6 +22,7 @@ const hasFocusVisibleSupport = hasDocument && supportsFocusVisible();
 // getter for easy mocking
 export const getHasFocusVisibleSupport = (): boolean => hasFocusVisibleSupport;
 
+// TODO: is this still needed?
```
Comment (denyo) [inline packages/components/src/styles/focus-visible-fallback.ts:25]: 🤷

**Comment by denyo** [comment]
Scroll buttons look misplaced now.
Bildschirm­foto 2023-02-06 um 15 29 07

**Comment by mhaegeleMHP** [comment]
> Scroll buttons look misplaced now. Bildschirm­foto 2023-02-06 um 15 29 07

Yes it does, same as scroll button on tags. This is caused due to a collision of our new font sizes and the button size. We need a smaller button. Its already documented on the v3 status update.

File: packages/components-js/tests/vrt/puppeteer/specs/component-scaling.vrt.ts [hunk 4]
```diff
@@ -43,6 +45,10 @@ it.each(components)('should have no visual regression for scaled component %s',
             fixed: 48,
           },
         });
+        if (component === 'tabs' || component === 'tabs-bar') {
+          // Position of status bar is calculated via JS and needs the reload to be calculated correctly
+          await page.reload({ waitUntil: 'networkidle0' });
```
[the remark this answers is not in the document]
Comment (denyo) [reply packages/components-js/tests/vrt/puppeteer/specs/component-scaling.vrt.ts:50]: Reloading reloads the page. What is different is that the CDP settings are applied now.  
In any case reloading seems like the least elegant solution to work around the actual problem.

File: packages/components/src/components/tabs/tabs/tabs-styles.ts [hunk 5]
```diff
@@ -9,7 +8,7 @@ export const getComponentCss = (): string => {
       },
     },
     root: {
-      marginBottom: pxToRemWithUnit(8),
+      marginBottom: '8px',
```
Comment (denyo) [inline packages/components/src/components/tabs/tabs/tabs-styles.ts:11]: `hostHiddenStyles` are missing

File: packages/shared/src/testing/vrt.ts [hunk 6]
```diff
@@ -84,6 +84,7 @@ export const getVisualRegressionPropTableTester = (): VisualRegressionTester =>
 
 type VRTestOptions = TestOptions & {
   scenario?: (page: Page) => Promise<void>;
+  scaledComponents?: boolean;
```
Comment (denyo) [inline packages/shared/src/testing/vrt.ts:87]: Can we rename this to something that sounds more like a boolean and from the perspective of configuring the test setup?

**Review by denyo — APPROVED** [review]

**Pull request merged**